content/en/blog/_posts/2023-07-24-unknown-version-interoperability-proxy-alpha.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: blog
+title: "Unknown Version Interoperability Proxy Alpha"
+date: 2023-05-24
+slug: kubernetes-1-28-feature-unknown-version-interoperability-proxy-alpha
+---
+
+**Authors:** Richa Banker (Google)
+
+This blog describes “Unknown Version Interoperability Proxy”, a new alpha feature in Kubernetes 1.28. Unknown Version Interoperability Proxy enables a resource request to be served by the correct apiserver in cases where there are multiple apiservers at varied version in a cluster, ex: during an upgrade/downgrade of a cluster, when a runtime config changes and a rollout occurs.
+
+## What problem does this solve?
+When a cluster undergoes an upgrade or downgrade, the kube-apiservers existing at different versions in that scenario can serve different sets (groups, versions, resources) of built-in resources. A resource request made in this scenario may be served by any of the available apiservers, potentially resulting in the request ending up at an apiserver that may not be aware of the requested resource; consequently it being served a 404 not found error which is incorrect.
+
+Furthermore, incorrect serving of the 404 errors can lead to serious consequences such as namespace deletion being blocked incorrectly or objects being garbage collected mistakenly.
+
+## How do we solve the problem?
+The new feature “Unknown Version Interoperability Proxy” provides the kube-apiserver with the capability to proxy a request to a peer kube-apiserver which is aware of the requested resource and hence can serve the request. To do this, a new filter is added to the handler chain in the aggregator which proxies clients to an apiserver that is capable of handling their request
+
+1. The existing StorageVersion API is used to figure out which group, versions, and resources an apiserver can serve
```

Comment:
Readers may consult https://kubernetes.io/docs/reference/kubernetes-api/ and wonder where StorageVersion went. Some people might think this is like StorageState from `kube-storage-version-migrator`.

We should give enough information to make it clear what kind of API StorageVersion is.

Reply:
Added a link to the StorageVersion API referred to in the text.